backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -295,25 +298,70 @@ def get_mf_permissions(entityUri):
+                        session,
+                        recipient=owner,
+                        target_uri=f'{attached.entityUri}|{attached.entityType}',
+                        message=f'New version {new_version.version} is available for metadata form "{mf.name}"',
+                        notification_type='METADATA_FORM_UPDATE',
+                    )
         return new_version.version
 
     @staticmethod
     @TenantPolicyService.has_tenant_permission(MANAGE_METADATA_FORMS)
     @MetadataFormAccessService.can_perform(UPDATE_METADATA_FORM_FIELD)
     def delete_metadata_form_version(uri, version):
         with get_context().db_engine.scoped_session() as session:
-            all_versions = MetadataFormRepository.get_metadata_form_versions(session, uri)
+            all_versions = MetadataFormRepository.get_metadata_form_versions_numbers(session, uri)
             if len(all_versions) == 1:
                 raise UnauthorizedOperation(
                     action='Delete version', message='Cannot delete the only version of the form'
                 )
             mf = MetadataFormRepository.get_metadata_form_version(session, uri, version)
             session.delete(mf)
             return MetadataFormRepository.get_metadata_form_version_number_latest(session, uri)
+
+    @staticmethod
+    def list_metadata_form_versions(uri):
+        with get_context().db_engine.scoped_session() as session:
+            all_versions = MetadataFormRepository.get_metadata_form_versions(session, uri)
+            for v in all_versions:
+                v.attached_forms = len(MetadataFormRepository.get_all_attached_metadata_forms(session, uri, v.version))
+            return all_versions
+
+    @staticmethod
+    def resolve_attached_entity(attached_metadata_form):
+        with get_context().db_engine.scoped_session() as session:
+            if attached_metadata_form.entityType == MetadataFormEntityTypes.Organizations.value:
+                return OrganizationRepository.get_organization_by_uri(session, attached_metadata_form.entityUri)
+            elif attached_metadata_form.entityType == MetadataFormEntityTypes.Environments.value:
+                return EnvironmentRepository.get_environment_by_uri(session, attached_metadata_form.entityUri)
+            elif attached_metadata_form.entityType == MetadataFormEntityTypes.Datasets.value:
+                return DatasetBaseRepository.get_dataset_by_uri(session, attached_metadata_form.entityUri)
+            else:
+                return None
+
+    @staticmethod
+    def get_entity_name(attached_metadata_form):
+        entity = MetadataFormService.resolve_attached_entity(attached_metadata_form)
+        return entity.name if entity else 'Not Found'
+
+    @staticmethod
+    def get_entity_owner(attached_metadata_form):
+        entity = MetadataFormService.resolve_attached_entity(attached_metadata_form)
+        if entity:
+            return entity.SamlGroupName or entity.SamlAdminGroupName
```

Comment:
get an error I believe stemming from here `'S3Dataset' object has no attribute 'SamlGroupName'` when creating a new version

this logic will raise an error - need to check if entity has the attribute `SamlGroupName` (prob can use hasattr(entity,  'SamlGroupName')` for this

Reply:
Heh. I've already found it and made a commit, but in different branch. 
Here it is now